docs/review.md

```diff
@@ -0,0 +1,124 @@
+# Code review
+
+This document aims to explain why code review can be a very valuable form of
+contribution to any Open Source project.
+But code review is a very broad term that can mean many things. Good in-depth
+code review is also a skill that needs to be learned but can be highly valuable
+in any development team.
+
+## Why should new contributors review code first?
+
+Developers interested in contributing to an Open Source project seem to often
+think that _writing code_ is the bottleneck in those projects.
+That rarely is the case though. Especially in Bitcoin related projects, where
+any mistake can have enormous consequences, any code that wants to be merged
+needs to go through thorough review first (often requiring two or more other
+developers to approve before it can be merged).
+
+Therefore, the actual bottleneck in Bitcoin related Open Source projects is
+**review, not the production of code**.
+
+So new contributors that come into a project by opening a PR often take more
+time away from the maintainers of that project than what their first
+contribution ends up adding in value. That is especially the case if the
+contributors don't have a lot of experience with git, GitHub, the target
+programming language and its formatting requirements or the review and test
+process as a whole.
+
+If a new contributor instead starts reviewing Pull Requests first, before
+opening their own ones, they can achieve the following advantages:
+ - They learn about the project, its formal requirements and the programming
+   language used, without anyone's assistance
+ - They learn about how to compile, execute and test the project they are aiming
+   to contribute to
+ - They see best practices in terms of formatting, naming, commit structure,
+   etc.
+ - The value added to the project by reviewing, running and testing code is
+   likely positive from day one and requires much less active assistance and
+   time from maintainers
+
+The maintainers of this project therefore ask all new contributors to review
+at least a couple of Pull Requests before adding their own.
+
+## How do I review code?
+
+Reviewing code in a constructive and helpful way is almost an art form by
+itself. The following list is definitely not complete and every developer tends
+to develop their own style when it comes to code review. But the list should
+have the most basic tasks associated with code review that should be a good
+entry point for anyone not yet familiar with the process:
+
+- Make sure to pick a Pull Request that fits your level of experience. You
```

Comment:
I'd also maybe mention something like: "If you're new to the area that you'd like to review, first familiarize yourself with the code, try to understand the architecture, public interfaces, interactions and code coverage. It can be useful to look at previous PRs to get more insight."

Reply:
Good idea, added!